### ISSUE TITLE
Document Package with custom roclet

### ISSUE THREAD

**Issue by mikldk** [rstudio/rstudio#5201]
I have implemented a custom `roxygen2` roclet ([`roxytest`](https://github.com/mikldk/roxytest)), and I have a project with a package using it, meaning that my `DESCRIPTION` contains this:
```
Roxygen: list(roclets = c("namespace", "rd", "roxytest::testthat_roclet"))
```

In Rstudio under Project Options under Build Tools I have enabled "Generate documentation with roxygen" and under Configure I can choose the four different `roxygen2` roclets (`rd`, `namespace`, `collate`, `vignette`) and then when to do it.

When I then trigger the Document Package (e.g. by `Ctrl+Shift+D`/`Cmd+Shift+D`) then 
```
devtools::document(roclets = c('rd', 'collate', 'namespace'))
```
is executed. Without the `roxytest::testtthat_roclet` specified in the `DESCRIPTION` file. That means that to use the custom `roxytest::testtthat_roclet` I need to manually execute 
```
roxygen2::roxygenise()
```

**My feature request is to somehow trigger (e.g. by `Ctrl+Shift+D`/`Cmd+Shift+D`) `devtools::document()` without specifying the `roclets` argument.**

This would make it more user-friendly to use custom roclets.

One way is to uncheck all roclets. But currently, then nothings happens when pressing `Ctrl+Shift+D`/`Cmd+Shift+D`. 

Another way is that under Project Options -> Build Tools -> Configure (next to "Generate documentation with roxygen") add an option "Use specification in DESCRIPTION file" (and then the four default `roxygen2` roclets cannot be chosen) -- or something like that.

**Comment by ronblum** [comment]
@mikldk Thank you for the suggestion! We'll review it as we continue development of RStudio.

**Comment by mikldk** [comment]
I was wondering if this could be fixed by removing [L89-92 in `src/gwt/src/org/rstudio/studio/client/projects/ui/prefs/buildtools/BuildToolsPackagePanel.java`](https://github.com/rstudio/rstudio/blob/d8a6a673c8ea6d85ab558b75b5a78525b13b19df/src/gwt/src/org/rstudio/studio/client/projects/ui/prefs/buildtools/BuildToolsPackagePanel.java#L89), i.e. removeing
```
                     chkUseRoxygen_.setValue(input.getRocletRd() || 
                                             input.getRocletCollate() || 
                                             input.getRocletNamespace() ||
                                             input.getRocletVignette());
```

**Comment by mikldk** [comment]
Any news on this?

**Comment by mikldk** [comment]
Note that the PR was merged into `devtools` in r-lib/devtools@919b3f474a82dcaf9bcabc0c60919ea5a0a424f3. This partly solves the problem (but requires the user to redefine keyboard shortcuts), but I still think it would be nice to try to fix it generally in Rstudio.

**Comment by ramiromagno** [comment]
I second this.
